src/Adapter/Product/AdminProductWrapper.php

```diff
@@ -496,19 +496,48 @@ public function processProductCustomization($product, $data)
 
         $shopList = Shop::getContextListShopID();
 
+        /** Update the customization fields to be deleted in the next step if not used */
+        $updateQuery = 'UPDATE `' . _DB_PREFIX_ . 'customization_field` cf
+            SET cf.`is_deleted` = 1
+            WHERE
+            cf.`id_product` = ' . (int)$product->id . ' 
+            AND cf.`is_deleted` = 0 ';
+        if (!empty($customization_ids)) {
+            $updateQuery .= 'AND cf.`id_customization_field` NOT IN (' . implode(',', array_map('intval', $customization_ids)) . ')';
+        }
+        Db::getInstance()->execute($updateQuery);
```

Comment:
Please do not build and/or execute SQL queries here. Adapters are not model classes.